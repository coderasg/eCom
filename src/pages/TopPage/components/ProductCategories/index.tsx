import React from 'react';

import useCategories from '../../../../hooks/fetchAPIs/useCategories';


import Tabs from '../../../../components/Tabs';
import Tab from '../../../../components/Tab';

import styles from './productCategories.module.scss';

type Props = {
  handleTabClick: (categoryId: number) => void;
  selectedId: number | null
};

function ProductCategories(props: Props) {
  const { handleTabClick, selectedId } = props;

  const { data: categories, error: categoryError } = useCategories();


  if (categoryError) {
    return (
      <h1 data-testid='ProductCategories-Error'>{`failed to load: ${categoryError}`}</h1>
    );
  }

  if (!categories) {
    return (
      <h1 data-testid='ProductCategories-Loading' className={styles.productCategoriesLoading}>{'Loadiding Categories.....'}</h1>
    );
  }

  return (
    <div data-testid="ProductCategories-wrapper">
      <Tabs>
        {categories.map((category) => (
          <Tab
            key={category.name}
            name={category.name}
            isSelected={Number(category.id) === selectedId}
            handleTabClick={() => {
              handleTabClick(Number(category.id));
            }}
          />
        ))}
      </Tabs>
    </div>
  );
}

export default ProductCategories;
