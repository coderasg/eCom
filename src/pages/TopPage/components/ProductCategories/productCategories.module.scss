.productCategoriesLoading{
    min-height: 10vh;
}