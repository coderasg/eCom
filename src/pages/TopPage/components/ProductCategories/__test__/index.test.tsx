import React from 'react';
import renderer from 'react-test-renderer';
import useSWR from 'swr';

import ProductCategories from '../index';

jest.mock('swr');

(useSWR as jest.Mock).mockReturnValue({
  data: {
    data: [
      {
        id: '1',
        name: 'インテリア',
      },
      {
        id: '2',
        name: 'レディース',
      },
      {
        id: '3',
        name: 'ベビー・キッズ',
      },
    ],
  },
});

describe('(Component) ProductCategories', () => {
  const defaultProps = {
    handleTabClick: (id: number) => {
      console.log('tab click', id);
    },
    selectedId: 1,
  };
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<ProductCategories {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
