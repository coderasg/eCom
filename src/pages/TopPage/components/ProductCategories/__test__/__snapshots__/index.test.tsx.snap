// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProductCategories should pass snapshot test 1`] = `
<div
  data-testid="ProductCategories-wrapper"
>
  <div
    data-testid="Tabs-wrapper"
  >
    <div
      className="undefined undefined"
      data-testid="Tab-wrapper"
      onClick={[Function]}
    >
      インテリア
      <SvgIcon
        fill="#f1f1f1"
        height={20}
        width={20}
      />
    </div>
    <div
      className="undefined false"
      data-testid="Tab-wrapper"
      onClick={[Function]}
    >
      レディース
    </div>
    <div
      className="undefined false"
      data-testid="Tab-wrapper"
      onClick={[Function]}
    >
      ベビー・キッズ
    </div>
  </div>
</div>
`;
